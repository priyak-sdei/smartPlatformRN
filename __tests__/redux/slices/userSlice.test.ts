import userReducer, { clearUser } from '../../../src/redux/slices/userSlice';

describe('userSlice', () => {
    it('should return the initial state', () => {
        expect(userReducer(undefined, { type: 'user/unknown' })).toEqual({
            email: null,
        });
    });

    it('should handle clearUser', () => {
        const previousState = { email: 'test@example.com' };
        expect(userReducer(previousState, clearUser())).toEqual({ email: null });
    });

    it('matches snapshot', () => {
        expect(userReducer(undefined, { type: 'user/unknown' })).toMatchSnapshot('userSlice initial state');
    });
}); 