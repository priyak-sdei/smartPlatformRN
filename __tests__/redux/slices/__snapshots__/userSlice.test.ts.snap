// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userSlice matches snapshot: userSlice initial state 1`] = `
{
  "email": null,
}
`;
